osbuild/solver/dnf.py

```diff
@@ -322,9 +338,83 @@ def depsolve(self, arguments):
             "solver": "dnf",
             "packages": packages,
             "repos": repositories,
+            "modules": {},
         }
 
         if "sbom" in arguments:
             response["sbom"] = self._sbom_for_pkgset(last_transaction)
 
+        # if any modules have been requested we add sources for these so they can
+        # be used by stages to enable the modules in the eventual artifact
+        modules = {}
+
+        for transaction in transactions:
+            if transaction.get("module-install-specs") or transaction.get("module-enable-specs"):
+                # we'll be checking later if any packages-from-modules are in the
+                # packages-to-install set so let's do this only once here
+                package_nevras = list(p["nevra"] for p in packages)
+
+                for module_spec in itertools.chain(
+                    transaction.get("module-install-specs", []),
+                    transaction.get("module-enable-specs", []),
+                ):
+                    module_packages, module_nsvcap = self.base_module.get_modules(module_spec)
```

Comment:
Here, I'd iterate over `module-enable-specs` and `install-specs` (I think modules must be prefixed with `@`) and call `self.base_module.get_modules()` on them to determine if they correspond to a module.